core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java

```diff
@@ -146,15 +172,25 @@ public String toString() {
       justification = "Field is written by Gson")
   @Override
   public void init(InitParameters params) {
-    ExecutorConfig[] execConfigs =
-        GSON.get().fromJson(params.getOptions().get("executors"), ExecutorConfig[].class);
-
     List<Executor> tmpExec = new ArrayList<>();
+    ExecutorConfig[] execConfigs = null;
+    QueueConfig[] queueConfigs = null;
+
```

Comment:
Could make the executors and queues options mutually exclusive, if both are seen then throw an exception. If they are not mutually exclusive then could throw an exception if they both define the same queue with differing max sizes.

Reply:
I want to ensure that the user is able to define internal compactions via the `executors` option and also add external compactors via `queues`. Would making these mutually exclusive break that functionality?

There is a check on the requested queue name already when the externalId is first requested for executor creation. 

https://github.com/apache/accumulo/blob/d811807bf1f6ce4f78cfb6237c995cedb6650038/core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionPlannerInitParams.java#L81-L84

The duplicate queue name failure condition is also written up as a test case: `CheckCompactionConfigTest#testRepeatedQueueName`

https://github.com/apache/accumulo/pull/3915/commits/6794f528db33ba0b8a1c06c1716895cb47393947#diff-784bf2d7143245058a1ee93ab53f84442106b03b036f3134dec83ee381981005R171-R187

Reply:
> I want to ensure that the user is able to define internal compactions via the executors option and also add external compactors via queues. Would making these mutually exclusive break that functionality?

I did not realize that was goal, I like that goal and my suggestion would break that.  I was thinking of the case of only using external  compactions, but being able to mix them and transition to the new property seems nice.  And there is handling of same queue name being specified in executors and queues, which is nice.